src/lib/connection.ts

```diff
@@ -94,6 +87,11 @@ export interface ConfirmOptions {
   rate?: number;
 }
 
+export type PrefixOptions = HashOptions | CreateOptions;
```

Comment:
Removed `MethodOptions` and opted for various discriminated unions that align to a corresponding `util.ts` function.